File: Backend/src/Nexus.Domain/Entities/Project.cs
namespace Nexus.Domain.Entities;

/// <summary>
/// Layihə (Project)
/// </summary>
public class Project
{
    public long ProjectId { get; set; }
    
    /// <summary>
    /// Unikal layihə kodu: PRJ-2024-001
    /// </summary>
    public string ProjectCode { get; set; } = null!;
    
    public string ProjectName { get; set; } = null!;
    public string? Description { get; set; }
    
    /// <summary>
    /// Təşkilat kodu (təcrid üçün)
    /// </summary>
    public string OrganizationCode { get; set; } = "default";
    
    /// <summary>
    /// Əlaqəli sənəd qovluğu (məktublar üçün)
    /// </summary>
    public long? DocumentNodeId { get; set; }
    
    public ProjectStatus Status { get; set; } = ProjectStatus.Active;
    public DateTime? StartDate { get; set; }
    public DateTime? EndDate { get; set; }
    
    // Audit
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public string CreatedBy { get; set; } = "system";
    public DateTime? ModifiedAt { get; set; }
    public string? ModifiedBy { get; set; }
    
    // Navigation
    public DocumentNode? DocumentNode { get; set; }
    public ICollection<TaskItem> Tasks { get; set; } = new List<TaskItem>();
    public ICollection<UserProjectRole> UserRoles { get; set; } = new List<UserProjectRole>();
}

public enum ProjectStatus
{
    Planning,
    Active,
    OnHold,
    Completed,
    Cancelled
}

File: mobile/lib/models/project.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'project.freezed.dart';
part 'project.g.dart';

@freezed
class Project with _$Project {
  const factory Project({
    required int projectId,
    required String projectCode,
    required String projectName,
    String? description,
    required String organizationCode,
    required ProjectStatus status,
    DateTime? startDate,
    DateTime? endDate,
    int? totalTasks,
    int? completedTasks,
    double? progressPercent,
    DateTime? createdAt,
  }) = _Project;

  factory Project.fromJson(Map<String, dynamic> json) =>
      _$ProjectFromJson(json);
}

enum ProjectStatus {
  planning,
  active,
  onHold,
  completed,
  cancelled,
}

@freezed
class ProjectSummary with _$ProjectSummary {
  const factory ProjectSummary({
    required int projectId,
    required String projectName,
    required ProjectStatus status,
    required int taskCount,
    required int completedTasks,
    required double progressPercent,
  }) = _ProjectSummary;

  factory ProjectSummary.fromJson(Map<String, dynamic> json) =>
      _$ProjectSummaryFromJson(json);
}
